spec/std/socket/udp_socket_spec.cr

```diff
@@ -52,6 +52,30 @@ describe UDPSocket do
       client.close
       server.close
     end
+
+    it "can join and transmit to multicast groups" do
+      udp = UDPSocket.new Socket::Family::INET
```

Comment:
There should also be a test for IPv6. A simple solution would be to use `family` here (it comes from `each_ip_family` in line 5). The any address (for the next line) is provided as fourth argument to the block by `each_ip_family`.

Only the lines for `multicast_interface` and goup handling need to be duplicated and distinguished between v4 and v6.

Reply:
Perhaps it would be better then to just duplicate the entire spec or split the familiy-specific parts to separate specs. Maybe that's even better to have smaller specs focused on single features.

Reply:
Ahh nice, didn't notice that. Will update.